server/apps/immich/src/api-v1/api-key/api-key.service.ts

```diff
@@ -0,0 +1,74 @@
+import { UserEntity } from '@app/database';
+import { BadRequestException, Inject, Injectable, UnauthorizedException } from '@nestjs/common';
+import { compareSync, hash } from 'bcrypt';
+import { randomBytes } from 'node:crypto';
+import { AuthUserDto } from '../../decorators/auth-user.decorator';
+import { IKeyRepository } from './api-key.repository';
+import { APIKeyCreateDto } from './dto/api-key-create.dto';
+import { APIKeyCreateResponseDto } from './repsonse-dto/api-key-create-response.dto';
+import { APIKeyResponseDto, mapKey } from './repsonse-dto/api-key-response.dto';
+
+@Injectable()
+export class APIKeyService {
+  constructor(@Inject(IKeyRepository) private repository: IKeyRepository) {}
+
+  async create(authUser: AuthUserDto, dto: APIKeyCreateDto): Promise<APIKeyCreateResponseDto> {
+    const key = randomBytes(24).toString('base64').replace(/\W/g, '');
+    const entity = await this.repository.create({
+      key: await hash(key, 10),
+      name: dto.name || 'API Key',
+      userId: authUser.id,
+    });
+
+    const secret = Buffer.from(`${entity.id}:${key}`, 'utf8').toString('base64');
+
+    return { secret, apiKey: mapKey(entity) };
+  }
+
+  async update(authUser: AuthUserDto, id: number, dto: APIKeyCreateDto): Promise<APIKeyResponseDto> {
+    const exists = await this.repository.getById(authUser.id, id);
```

Comment:
just nitpicking here since we are seeing we are doing this more often, can we refactor the `getById` to another method of this service, since it is reused a lot?